simulariumio/trajectory_converter.py

```diff
@@ -148,6 +149,35 @@ def get_subpoints_xyz(subpoints: np.array, n_subpoints: np.array) -> np.array:
                 )
         return xyz_subpoints.reshape(1, -1, 3)
 
+    @staticmethod
+    def translate_positions(data: AgentData, translation: np.ndarray) -> AgentData:
+        total_steps = data.times.size
+        max_subpoints = int(np.amax(data.n_subpoints))
+        for time_index in range(total_steps):
+            for agent_index in range(int(data.n_agents[time_index])):
+                display_type = data.display_type_for_agent(time_index, agent_index)
+                translate_subpoints = (
+                    max_subpoints > 0 and display_type != DISPLAY_TYPE.SPHERE_GROUP
+                )
+                if translate_subpoints:
+                    sp_items = Filter.get_items_from_subpoints(
+                        data, time_index, agent_index
+                    )
+                    if sp_items is None:
+                        translate_subpoints = False
+                    else:
+                        # translate subpoints for fibers
+                        n_items = sp_items.shape[0]
+                        for item_index in range(n_items):
+                            sp_items[item_index][:VALUES_PER_3D_POINT] += translation
+                        n_sp = int(data.n_subpoints[time_index][agent_index])
+                        data.subpoints[time_index][agent_index][
+                            :n_sp
+                        ] = sp_items.reshape(n_sp)
+                if not translate_subpoints:
+                    data.positions[time_index][agent_index] += translation
```

Comment:
I'm also confused why you wouldn't do this line no matter what if they have subpoints. I'm sure it's working, I think it could just use more comments and explanations

Reply:
I can't remember exactly, but I don't think the viewer handles local position offsets well for fibers, so we leave positions at zero and only use subpoint positions. @ascibisz it would be good to test the current viewer with fibers with non-zero positions to check this is still the case, we can chat about it tomorrow.

Reply:
Please put an issue in simularium-viewer if it is true that we can't position fibers in this way.

Reply:
Ok so I played around in the simularium viewer and the viewer just ignores the agent data position for fibers! So the if statement here _can_ be removed, even though we technically don't need to move the agent position, it doesn't really matter if we do. I think it's less confusing if we just translate it anyways, so I'll do that

Reply:
- [ ] remove if statement for translating agent position

Reply:
Just to make sure I understand, if the agent position for a fiber is not (0,0,0), then shouldn't the subpoints be relative to that position?

Reply:
One more thought to add:  if the data is generated so that every fiber is at position 0,0,0 and rotation 0,0,0, and the subpoints are baked to the right positions, the viewer should work and people can continue making data like that.  I'm just saying that the fact that we don't account for position and rotation in the viewer should be an issue ticket!  I want to revisit that and make sure we fully understand the technical limitation and fix it if we can.

Reply:
Maybe we should discuss this more during a dev chat, since I really haven't dug very much into the viewer code to know how all of this works.  What I do know is that when we generate data for fibers, the agent position is always (0, 0, 0) and all of the actual position information is contained in the subpoints. So I guess technically the subpoints are relative to the agent position, in the same way that any position data is relative to the origin! But yeah if you change the agent position and leave the subpoints the same, it will look the same. I haven't played around with rotations much, but I'll do that now

Also for sphere groups, the subpoint positions are relative to the agent position though, so that's fun.

Reply:
I'm sure you're right but I just think we should have the data output in a way that is at least consistent with being able to rotate and translate any agent.(So keep the center at 0,0,0)  I'm not going to suggest changing the way we output fibers until we know how hard it will be to make the viewer handle it!  Also, if the idea of having a center and rotation for a fiber is a concept most engines don't deal with anyway, then it could be a moot point.

Reply:
Ok @blairlyons let's talk about this during 1:1 today, but I'll probably just revert the commit where I removed the if statement that kept the agent position for fibers at (0, 0, 0)

Reply:
alright @toloudis I switched back to keeping the agent position at the origin when we translate fiber agents that have subpoints, because at least that's consistent with how we've historically translated agents.

I also cut a [ticket](https://github.com/simularium/simularium-viewer/issues/332) for simularium-viewer to look into this more and likely switch to interpreting fiber subpoints as relative to the agent position, like we do for sphere groups!